lib/tts_isolate.dart

```diff
@@ -24,14 +27,39 @@ Future<void> isolateMain(
   await Firebase.initializeApp();
   final ttsQueue = TTSQueue();
 
-  // listen for changes to the tts preferences
-  bool isPreludeMuted = false;
+  final ttsModel = TtsModel.fromJson(
+      jsonDecode(prefs.getString("tts", defaultValue: '{}').getValue()));
+
+  // listen for changes to the tts preferences and update the isolates ttsModel
   final ttsPrefs = prefs.getString('tts', defaultValue: '{}');
-  ttsPrefs.listen((value) {
-    final Map<String, dynamic> ttsMap = jsonDecode(value);
-    isPreludeMuted = ttsMap['isPreludeMuted'];
+  ttsPrefs.listen((value) async {
+    ttsModel.updateFromJson(jsonDecode(value));
+    await TextToSpeechPlugin.updateTTSPreferences(
+        ttsModel.pitch, ttsModel.speed);
   });
 
+  TwitchMessageModel createMessageModelFromData(
+      String messageId, Map<String, dynamic> messageData, DateTime timestamp) {
+    return TwitchMessageModel(
```

Comment:
This mapping is almost correct, you can simplify it a little bit. Steal the code from here: https://github.com/muxable/rtchat/blob/96b8798af1db8cd73c5197d927ece0eae951680e/lib/models/adapters/messages.dart#L97-L108